### ISSUE TITLE
issue with kind0.7.0, Error invoking Delegate Add missing network name

### ISSUE THREAD

**Issue by hyson007** [networkop/meshnet-cni#20]
test env: kind0.7.0 with default kindnet
meshnet-cni manifest applied:  meshnet-cni/manifests/base/meshnet.yml

pod stuck at ContainerCreating

`Warning  FailedCreatePodSandBox  23s (x17 over 3m48s)  kubelet, kindnet-worker2  (combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "645a8daac55cc54853a1be8ff1e4db675505c3e55585f55f1660d65d106e2869": Error invoking Delegate Add missing network name`

journalctl from worker2 node

```
Mar 21 02:19:09 kindnet-worker2 kubelet[539]: I0321 02:19:09.457259     539 reconciler.go:209] operationExecutor.VerifyControllerAttachedVolume started for volume "default-token-7nvc2" (UniqueName: "kubernetes.io/secret/a8c8f3cd-4f
3e-476a-8c0d-7f94f246a1ac-default-token-7nvc2") pod "r1" (UID: "a8c8f3cd-4f3e-476a-8c0d-7f94f246a1ac")
Mar 21 02:19:09 kindnet-worker2 kubelet[539]: E0321 02:19:09.677373     539 remote_runtime.go:105] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = failed to setup network for sandbox "4ba52ab25c5a31e7177
248e5122b4c8cdeaae14e060d1a67da3de28f534498bb": Error invoking Delegate Add missing network name
Mar 21 02:19:09 kindnet-worker2 kubelet[539]: E0321 02:19:09.677472     539 kuberuntime_sandbox.go:68] CreatePodSandbox for pod "r1_default(a8c8f3cd-4f3e-476a-8c0d-7f94f246a1ac)" failed: rpc error: code = Unknown desc = failed to s
etup network for sandbox "4ba52ab25c5a31e7177248e5122b4c8cdeaae14e060d1a67da3de28f534498bb": Error invoking Delegate Add missing network name
Mar 21 02:19:09 kindnet-worker2 kubelet[539]: E0321 02:19:09.677495     539 kuberuntime_manager.go:729] createPodSandbox for pod "r1_default(a8c8f3cd-4f3e-476a-8c0d-7f94f246a1ac)" failed: rpc error: code = Unknown desc = failed to
setup network for sandbox "4ba52ab25c5a31e7177248e5122b4c8cdeaae14e060d1a67da3de28f534498bb": Error invoking Delegate Add missing network name
Mar 21 02:19:09 kindnet-worker2 kubelet[539]: E0321 02:19:09.677614     539 pod_workers.go:191] Error syncing pod a8c8f3cd-4f3e-476a-8c0d-7f94f246a1ac ("r1_default(a8c8f3cd-4f3e-476a-8c0d-7f94f246a1ac)"), skipping: failed to "Creat
ePodSandbox" for "r1_default(a8c8f3cd-4f3e-476a-8c0d-7f94f246a1ac)" with CreatePodSandboxError: "CreatePodSandbox for pod \"r1_default(a8c8f3cd-4f3e-476a-8c0d-7f94f246a1ac)\" failed: rpc error: code = Unknown desc = failed to setup
 network for sandbox \"4ba52ab25c5a31e7177248e5122b4c8cdeaae14e060d1a67da3de28f534498bb\": Error invoking Delegate Add missing network name"
```

/etc/cni/net.d/

```
-rw-r--r-- 1 root root  411 Mar 21 02:17 00-meshnet.conf
-rw-r--r-- 1 root root  391 Mar 21 02:17 10-kindnet.conflist
-rw-r--r-- 1 root root  326 Mar 21 02:17 meshnet.conf

{
  "cniVersion": "0.2.0",
  "name": "meshnet_network",
  "type": "meshnet",
  "delegate": {
    "type": "ptp",
    "ipMasq": false,
    "ipam": {
      "type": "host-local",
      "dataDir": "/run/cni-ipam-state",
      "routes": [
        {
          "dst": "0.0.0.0/0"
        }
      ],
      "ranges": [
        [
          {
            "subnet": "10.244.2.0/24"
          }
        ]
      ]
    }
  }
}

{
        "cniVersion": "0.3.1",
        "name": "kindnet",
        "plugins": [
        {
                "type": "ptp",
                "ipMasq": false,
                "ipam": {
                        "type": "host-local",
                        "dataDir": "/run/cni-ipam-state",
                        "routes": [
                                {
                                        "dst": "0.0.0.0/0"
                                }
                        ],
                        "ranges": [
                        [
                                {
                                        "subnet": "10.244.2.0/24"
                                }
                        ]
                ]
                }
        },
        {
                "type": "portmap",
                "capabilities": {
                        "portMappings": true
                }
        }
        ]
}
{
  "cniVersion": "0.2.0",
  "name": "meshnet_network",
  "type": "meshnet",
  "delegate": {
    "name": "dind0",
    "bridge": "dind0",
    "type": "bridge",
    "isDefaultGateway": true,
    "ipMasq": true,
    "ipam": {
      "type": "host-local",
      "subnet": "10.244.1.0/24",
      "gateway": "10.244.1.1"
    }
  }
}
```

**Comment by networkop** [comment]
can you try it with kind version 0.3.0 instead? the newer version has CNI conflist (vs simple conf in 0.3.0), which i haven't got around to implementing yet.

**Comment by hyson007** [comment]
thanks for the quick response, yes, it works with kind 0.3.0.

**Comment by networkop** [comment]
fixed in #22
